src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+
+import usersFromServer from '../../api/users';
+
+import { getUserById } from '../../services/user';
+import { TodosWithUser } from '../../types/TodosWithUser';
+
+type Props = {
+  onSubmit: (todo: TodosWithUser) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+
+    setHasTitleError(false);
+    setHasUserError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      title,
+      completed: false,
+      id: 0,
+      userId,
+      user: getUserById(userId),
```

Comment:
Ensure that the `getUserById` function returns a valid user object or null, as expected by the `TodosWithUser` type definition.